stock_request_submit/models/stock_request.py

```diff
@@ -2,17 +2,17 @@
 # Copyright 2019-2020 ForgeFlow S.L. (https://www.forgeflow.com)
 # License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
 
-from odoo import fields, models
+from odoo import _, fields, models
 
 
 class StockRequest(models.Model):
     _inherit = "stock.request"
 
-    def __get_request_states(self):
-        states = super().__get_request_states()
-        if not ("submitted", "Submitted") in states:
+    def _get_request_states(self):
+        states = super()._get_request_states()
+        if not ("submitted", _("Submitted")) in states:
             states.insert(
-                states.index(("draft", "Draft")) + 1, ("submitted", "Submitted")
+                states.index(("draft", _("Draft"))) + 1, ("submitted", _("Submitted"))
```

Comment:
In any case, if this is the reason of this implementation, there's already a way to it without such overhead:

https://www.odoo.com/documentation/16.0/developer/reference/backend/orm.html?highlight=selection_add#odoo.fields.Selection
```
selection = [('a', 'A'), ('b', 'B')]
selection_add = [('c', 'C'), ('b',)]
> result = [('a', 'A'), ('c', 'C'), ('b', 'B')]
```

Another issue is that you need to mantain the uninstall hook, wich is also covered with `ondelete` parameter.

Reply:
Ok, then :sweat_smile:

Reply:
Ok, changes done using this approach (`selection_add`) and simplifying everything.